src/azure-cli-core/azure/cli/core/breaking_change.py

```diff
@@ -0,0 +1,596 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import abc
+import argparse
+from collections import defaultdict
+
+from knack.log import get_logger
+from knack.deprecation import Deprecated
+from knack.util import StatusTag, color_map
+
+
+logger = get_logger()
+
+NEXT_BREAKING_CHANGE_RELEASE = '2.67.0'
+DEFAULT_BREAKING_CHANGE_TAG = '[BrkChange]'
+
+
+def _get_action_class(cli_ctx, action):
+    action_class = argparse.Action
+
+    # action is either a user-defined Action class or a string referring a library-defined Action
+    if isinstance(action, type) and issubclass(action, argparse.Action):
+        action_class = action
+    elif isinstance(action, str):
+        action_class = cli_ctx.invocation.command_loader.cli_ctx.invocation.parser._registries['action'][action]  # pylint: disable=protected-access
+    return action_class
+
+
+def _argument_breaking_change_action(cli_ctx, status_tag, action):
+
+    action_class = _get_action_class(cli_ctx, action)
+
+    class ArgumentBreakingChangeAction(action_class):
+
+        def __call__(self, parser, namespace, values, option_string=None):
+            if not hasattr(namespace, '_argument_deprecations'):
+                setattr(namespace, '_argument_deprecations', [])
+            if isinstance(status_tag, MergedStatusTag):
+                for tag in status_tag.tags:
+                    namespace._argument_deprecations.append(tag)
+            else:
+                if status_tag not in namespace._argument_deprecations:
+                    namespace._argument_deprecations.append(status_tag)
+            try:
+                super().__call__(parser, namespace, values, option_string)
+            except NotImplementedError:
+                setattr(namespace, self.dest, values)
```

Comment:
What's this error catch used for?

Reply:
These lines are copied from Knack. I’m not sure why an action might have an unimplemented `__call__`, but a breaking change action could encounter a similar situation.